### PR TITLE
Support hashing a certificate directory

### PR DESCRIPTION
**Pull request by jannejy**
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

**Comment by jannejy** [comment]
> This doesn't look good to me. The behavior has been changed to only return the directory in which the bundle of roots is located. This will break all existing usage that depend on the file that contains the latest valid root, since not everyone is using directories.

Actually, not only directory, that is also checked in the tests here.

> Modifying code from existing functionality is never advised.
> 
> However, if a separate function that is clear and returns the location instead the bundle file is required, then it's better to create a new one in order to preserve backwards compatibility.

Anyway, I agree, that the changing of the interface is anot good idea. I'll add the implementaion for the location (as well as files as directories).

**Comment by AssemblyJohn** [comment]
Now it looks good, however make sure that in libocpp code, you are handling both the case of a directory and of a bundle file.

File: include/evse_security/evse_security.hpp [hunk 1]
```diff
@@ -227,6 +227,13 @@ class EvseSecurity {
     /// @return CA certificate file
     std::string get_verify_file(CaCertificateType certificate_type);
 
+    /// @brief Retrieves the PEM formatted CA bundle location for the given \p certificate_type It is not recommended to
+    /// add the SUBCAs to any root certificate bundle, but to leave them in the leaf file. Returns either file
+    /// or directory where the cerificates are located
```
Comment (barsnick) [inline include/evse_security/evse_security.hpp:232]: Typo: certificates
```suggestion
    /// or directory where the certificates are located
```

**Comment by jannejy** [comment]
The build fails not because of my changes, something is wrong after gmock installation

**Comment by jannejy** [comment]
> Does it work locally?

Yes

**Comment by AssemblyJohn** [comment]
When trying to build locally with: `cmake -DBUILD_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=./dist ..` I get a similar error. Looking into this.

**Comment by AssemblyJohn** [comment]
Rebase and try again, since we had a recent merge.

**Pull request merged**